### PR TITLE
Update test logging/exiting

### PR DESCRIPTION
**Pull request by emmawu**
Updated test logging
- log_warn() - log warn message and add warn count.
- log_error() - log error message and exit current test as FAIL.
- log_fatal() - log error message and exit current test as FAIL. All other tests will be aborted
- Current errors in all test cases are updated to non-fatal errors.

Wrapped calling of each test in run_test()
- No need to log "- Start" explicitly
- No need to call ready_to_exit() for each test explicitly

Updated test exiting
- run_test() calls ready_to_exit() at the end of each test
- If error/fatal error occurs, ready_to_exit() will be called immediately.

Signed-off-by: xiawu <xiawu@redhat.com>

### CODE REVIEW

File: lib/log.sh [hunk 1]
```diff
@@ -90,69 +96,80 @@ log_info()
...
-# @usage: log_fatal_error <mesg>
-# @description: log ERROR message and exit
+# @usage: log_fatal<mesg>
+# @description: log ERROR message
+#               and abort recipeset (only if beaker env)
 # @param1: mesg
-log_fatal_error()
+log_fatal()
 {
     log "ERROR" "$@"
```
Comment (lulinqing) [inline lib/log.sh]: should it be FATAL than ERROR here?

Comment (emmawu) [inline lib/log.sh]: Here the 'ERROR' is actually the level of error messages.  We only support  INFO / WARN/ERROR so far.  But I can add an extra level of error messages if that makes sense.

Comment (emmawu) [inline lib/log.sh]: Updated 2 methods in lib/log.sh:
- log()
- log_fatal() 
Pls review the latest version.

**Comment by sztsian** [comment]
@lulinqing I prefer to revoke my merging permission in case I do something wrong.

**Pull request merged**